runtime/shared_common/shrinit.cpp

```diff
@@ -3157,6 +3159,21 @@ j9shr_init(J9JavaVM *vm, UDATA loadFlags, UDATA* nonfatal)
 		*nonfatal = 1;
 	}
 
+	if (RESULT_DO_HELP != parseResult && RESULT_DO_MORE_HELP != parseResult) {
+		if (-1 == SH_OSCache::getCacheDir(vm, ctrlDirName, cacheDirName, J9SH_MAXPATH, cacheType)) {
```

Comment:
I'd prefer not to add another call to getCacheDir() in the startup path. The code should be optimal to provide the fastest startup possible.

Reply:
This code is removed.